src/tracer/MARBL_tracers.F90

```diff
@@ -0,0 +1,1387 @@
+  if (CS%tracer_inds%donr_ind > 0) &
+    CS%STF(:,:,CS%tracer_inds%donr_ind) = CS%STF(:,:,CS%tracer_inds%donr_ind) + fluxes%MARBL_forcing%donr_riv_flux(:,:)
+  if (CS%tracer_inds%dop_ind > 0) &
+    CS%STF(:,:,CS%tracer_inds%dop_ind) = CS%STF(:,:,CS%tracer_inds%dop_ind) + fluxes%MARBL_forcing%dop_riv_flux(:,:)
+  if (CS%tracer_inds%dopr_ind > 0) &
+    CS%STF(:,:,CS%tracer_inds%dopr_ind) = CS%STF(:,:,CS%tracer_inds%dopr_ind) + fluxes%MARBL_forcing%dopr_riv_flux(:,:)
+  if (CS%tracer_inds%sio3_ind > 0) &
+    CS%STF(:,:,CS%tracer_inds%sio3_ind) = CS%STF(:,:,CS%tracer_inds%sio3_ind) + fluxes%MARBL_forcing%sio3_riv_flux(:,:)
+  if (CS%tracer_inds%fe_ind > 0) &
+    CS%STF(:,:,CS%tracer_inds%fe_ind) = CS%STF(:,:,CS%tracer_inds%fe_ind) + fluxes%MARBL_forcing%fe_riv_flux(:,:)
+  if (CS%tracer_inds%doc_ind > 0) &
+    CS%STF(:,:,CS%tracer_inds%doc_ind) = CS%STF(:,:,CS%tracer_inds%doc_ind) + fluxes%MARBL_forcing%doc_riv_flux(:,:)
+  if (CS%tracer_inds%docr_ind > 0) &
+    CS%STF(:,:,CS%tracer_inds%docr_ind) = CS%STF(:,:,CS%tracer_inds%docr_ind) + fluxes%MARBL_forcing%docr_riv_flux(:,:)
+  if (CS%tracer_inds%alk_ind > 0) &
+    CS%STF(:,:,CS%tracer_inds%alk_ind) = CS%STF(:,:,CS%tracer_inds%alk_ind) + fluxes%MARBL_forcing%alk_riv_flux(:,:)
+  if (CS%tracer_inds%alk_alt_co2_ind > 0) &
+    CS%STF(:,:,CS%tracer_inds%alk_alt_co2_ind) = CS%STF(:,:,CS%tracer_inds%alk_alt_co2_ind) + &
+                                                 fluxes%MARBL_forcing%alk_alt_co2_riv_flux(:,:)
+  if (CS%tracer_inds%dic_ind > 0) &
+    CS%STF(:,:,CS%tracer_inds%dic_ind) = CS%STF(:,:,CS%tracer_inds%dic_ind) + fluxes%MARBL_forcing%dic_riv_flux(:,:)
+  if (CS%tracer_inds%dic_alt_co2_ind > 0) &
+    CS%STF(:,:,CS%tracer_inds%dic_alt_co2_ind) = CS%STF(:,:,CS%tracer_inds%dic_alt_co2_ind) + &
+                                                 fluxes%MARBL_forcing%dic_alt_co2_riv_flux(:,:)
+
+  ! (2) Post surface fluxes and their diagnostics (currently all 2D)
+  do m=1,CS%ntr
+    if (CS%id_surface_flux_out(m) > 0) &
+      call post_data(CS%id_surface_flux_out(m), CS%STF(:,:,m), CS%diag)
+  end do
+  do m=1,size(CS%surface_flux_diags)
+    if (CS%surface_flux_diags(m)%id > 0) &
+      call post_data(CS%surface_flux_diags(m)%id, CS%surface_flux_diags(m)%field_2d(:,:), CS%diag)
+  end do
+
+  ! (3) Apply surface fluxes via vertical diffusion
+  if (present(evap_CFL_limit) .and. present(minimum_forcing_depth)) then
+    do m=1,CS%ntr
+      do k=1,nz ;do j=js,je ; do i=is,ie
+        h_work(i,j,k) = h_old(i,j,k)
+      enddo ; enddo ; enddo
+      call applyTracerBoundaryFluxesInOut(G, GV, CS%tr(:,:,:,m) , dt, fluxes, h_work, &
+          evap_CFL_limit, minimum_forcing_depth)
+      call tracer_vertdiff(h_work, ea, eb, dt, CS%tr(:,:,:,m), G, GV, sfc_flux=CS%STF(:,:,m))
+    enddo
+  else
+    do m=1,CS%ntr
+      call tracer_vertdiff(h_old, ea, eb, dt, CS%tr(:,:,:,m), G, GV, sfc_flux=CS%STF(:,:,m))
```

Comment:
I think the units of `CS%STF are `CU * m * s-1`, where CU is the tracer concentration units. 
It looks like `tracer_vertdiff` expects `sfc_flux` to have units `[CU kg m-2 T-1 ~> CU kg m-2 s-1]`.
So I think we need a unit conversion that effectivley multiplies by density. The only tracer inside MOM that I see pass `sfc_flux` to `tracer_vertdiff` is in MOM_OCMIP2_CFC.F90. The comments at https://github.com/NOAA-GFDL/MOM6/blob/932b35c8c8d468370f7a243b26b0bb5651155cc7/src/tracer/MOM_OCMIP2_CFC.F90#L457-L463 describe this change of units to surface fluxes.

Reply:
Also note that https://github.com/NOAA-GFDL/MOM6/issues/1415 leads to an incorrect incorporation of `sfc_flux` in `tracer_vertdiff`. In short runs with a new implementation of CFCs, this parentheses bug leads to really crazy tracer values when `sfc_flux` is properly multiplied by density.  I expect crazy results in MARBL too if the units of `sfc_flux` are corrected and the parentheses bug is still there. We will need to include the fix to the parentheses bug before being able to vet MARBL solutions.